api/v1/submissions/PKPSubmissionHandler.php

```diff
@@ -517,14 +517,17 @@ public function add(SlimRequest $slimRequest, APIResponse $response, array $args
         } else {
             $submitAsUserGroup = Repo::userGroup()->getFirstSubmitAsAuthorUserGroup($context->getId());
             if (!$submitAsUserGroup) {
-                return $response->withStatus(400)->withJson([
-                    'userGroupId' => [__('submission.wizard.notAllowed.description')]
-                ]);
+                $errors[$userGroupIdPropName] = [__('submission.wizard.notAllowed.description')];
+            } else {
+                Repo::userGroup()->assignUserToGroup(
+                    $user->getId(),
+                    $submitAsUserGroup->getId()
+                );
             }
-            Repo::userGroup()->assignUserToGroup(
-                $user->getId(),
-                $submitAsUserGroup->getId()
-            );
+        }
+
+        if (!empty($errors)) {
```

Comment:
I am checking the errors array at the end of the checks so that the response can have all the possible errors